packages/runtime/container-runtime/src/summary/documentSchema.ts

```diff
@@ -0,0 +1,535 @@
+ * If it can't, it should not continue with document and immidiatly fail, preventing random (cryptic) failures
+ * down the road and potentially corrupting documents.
+ * For now this structure and appropriate interpretation / behavior is focused only on runtime features.
+ * In the future that could be interpolated to more areas, including DDSs used, and even possibly - application
+ * schema.
+ *
+ * Runtime will ignore any properties at the root that it does not understand (i.e. IDocumentSchema.app), but will
+ * stop (and fail session) on any unknown properties within "runtime" sub-tree.
+ *
+ * In most cases values preserved in the document will not dictate if such features should be enabled in a given session.
+ * I.e. if compression is mentioned in document schema, this means that runtime version that opens such document must know
+ * how to interpret such ops, but does not need to actually use compression itself. That said, some options could be
+ * sticky, i.e. influece feature selection for all runtimes openning a document. ID compression is one such example.
+ * Currently there is no mechanism to remove feature from this property bag, i.e. once compression was used, even if it's
+ * dissbled (through feature gate or code deployment), all existing documents that used compression will continue to fail
+ * if opened by clients who do not support compression.
+ *
+ * For now we are limitting it to just plain properties, and only really simple types, but that can be changed in the future.
+ *
+ * @alpha
+ */
+export interface IDocumentSchema {
+	// version that describes how data is stored in this structure.
+	// If runtime sees a version it does not understand, it should immidiatly fail and not
+	// attempt to interpret any further dafa.
+	version: string;
+
+	// Sequence number when this schema became active.
+	refSeq: number;
+
+	runtime: Record<string, DocumentSchemaValueType>;
+}
+
+/**
+ * Content of the type=ContainerMessageType.DocumentSchemaChange ops.
+ * The meaning of refSeq field is different in such messages (compared to other usages of IDocumentSchemaCurrent)
+ * ContainerMessageType.DocumentSchemaChange messages use CAS (Compare-and-swap) semantics, and convey
+ * regSeq of last known schema change (known to a client proposing schema change).
+ * @see ContainerRuntimeDocumentSchemaMessage
+ * @alpha
+ */
+export type IDocumentSchemaChangeMessage = IDocumentSchema;
+
+/**
+ * Current version known properties that define document schema
+ * This must be bumped whenever the format of document schema or protocol for changing the current document schema changes.
+ * Ex: adding a new configuration property (under IDocumentSchema.runtime) does not require changing this version.
+ * Ex: Changing the 'document schema acceptance' mechanism from convert-and-swap to one requiring consensus does require changing this version.
+ * @alpha
+ */
+export const currentDocumentVersionSchema = "1.0";
```

Comment:
No, it should not.  It may stay forever "1.0". Most likely reason we will switch to "2.0" if we want to change how consumers change schema (for example, move away from custom ops and use quorum, or something like that).
It would never become 1.1.
It' similar to summaryFormatVersion, and there we use numbers, so I'll switch to numbers here (i.e. 1, 2, etc.).

Reply:
Gotcha. Thanks, this makes more sense. I thought that adding any new stuff to the schema (such as a new type of compression) would need to be a versioned change wrt to this schema.